logging_loki/handlers.py

```diff
@@ -86,3 +79,38 @@ def emit(self, record: logging.LogRecord):
             self.emitter(record, self.format(record))
         except Exception:
             self.handleError(record)
+
+    def emit_batch(self, records: list[logging.LogRecord]):
+        """Send a batch of log records to Loki."""
+        # noinspection PyBroadException
+        try:
+            self.emitter.emit_batch([(record, self.format(record)) for record in records])
+        except Exception:
+            for record in records:
+                self.handleError(record)
+
+class LokiBatchHandler(MemoryHandler):
+    interval: float # The interval at which batched logs are sent in seconds
+    _last_flush_time: float
+    target: LokiHandler
+
+    def __init__(self, interval: float, capacity: int = LOKI_MAX_BATCH_BUFFER_SIZE, **kwargs):
+        super().__init__(capacity, **kwargs)
+        self.interval = interval
+        self._last_flush_time = time.time()
+
+    def flush(self) -> None:
+        self.acquire()
```

Comment:
does this acquired lock imply that all  logs made during the flush are lost because of the 
```
if not self._lock.acquire(blocking=False):
            return
```
in `__call__`?

Reply:
NVM, it's a lock on another object ... (the emitter, not the handler)

Reply:
Indeed, this lock is to prevent the entire buffer of logs to be flushed/sent twice

The other acquire in the emitter is to prevent the emitter from creating an infinite loop: 
when it's forwarding a log line to loki, we don't want to forward any newly created log lines in the emit methods itself (e.g. from urrlib/httpx/requests because of the request to loki)